### PR TITLE
teraslice-state-storage changes

### PR DESCRIPTION
**Pull request by ciorg**
* refactored to simplify the process and make more clear
* removes ability to pass custom function to sync (removes sync completely)
* added metaKey to allow for job to specify meta field to key on
* added documentation
* for some reason package.json for other packages rearranged when I ran bump script
* will also require a small change to elasticsearch-asset's elasticsearch -state-storage schema and config

### CODE REVIEW

File: docs/packages/teraslice-state-storage/overview.md [hunk 1]
```diff
@@ -4,3 +4,279 @@ sidebar_label: Overview
 ---
 
 > State storage operation api for teraslice
+
+Teraslice State Storage provides an LRU caching system, based on [mnemonist's](https://www.npmjs.com/package/mnemonist) LRU map, for teraslice processors. The in memory cache can be backed by a persistant storage system like Elasticsearch if a more robust cache is needed.
+
+The advantage of having the LRU cache backed by a persistant storage system like Elasticsearch is that if the key is not in the cache the processor will search an elasticsearch index for the key and if it is found will add it to the cache.  This essentially expands the cashe to the size of the underlying elasticsearch index without requiring the same memory resources in Teraslice.  The potential drawback is that on data sets with a large key set the processor will be continously seaching elasticsearch for each key which would render the caching mechanism pointless.
```
Comment (godber) [inline docs/packages/teraslice-state-storage/overview.md:10]: cashe -> cache

File: docs/packages/teraslice-state-storage/overview.md [hunk 2]
```diff
@@ -4,3 +4,279 @@ sidebar_label: Overview
...
+
+## Usage
+`Add the state storoge api must be added to the job file under the 'apis' property to be accessible by a processor.`
```
Comment (godber) [inline docs/packages/teraslice-state-storage/overview.md:13]: You have a lot of these `backtick` quoted things, have you looked to see how these get rendered?  Is this what you want?

https://github.com/terascope/teraslice/blob/f48deb03b1a94006f80841d0c07f6aea432223d8/docs/packages/teraslice-state-storage/overview.md

Comment (godber) [reply docs/packages/teraslice-state-storage/overview.md:13]: I can see using them around your function signatures, but not the explanatory text.

Comment (ciorg) [reply docs/packages/teraslice-state-storage/overview.md:13]: hmmm, yeah they look different in my editor.  I'll pull those out.

**Pull request merged**